src/Filter/StringValue.php

```diff
@@ -42,4 +38,35 @@ protected function doesMatch( $dataSet ) {
 		}
 		return false;
 	}
+
+	/**
+	 * Case-insensitive optional string comparison function
+	 *
+	 * @param string $operation - comparison option
+	 * @param string $haystack
+	 * @param string $needle
+	 * @return bool
+	 */
+	private function compareStrings( string $operation, string $haystack, string $needle ): bool {
+		$haystack = mb_strtolower( $haystack );
+		$needle = mb_strtolower( $needle );
+
+		switch ( $operation ) {
+			case self::COMPARISON_STARTS_WITH:
+				return str_starts_with( $haystack, $needle );
```

Comment:
Can not use `str_starts_with`/`str_ends_with`. It is PHP 8 only. We still need to be backwards compatible to PHP 7.4: https://www.php.net/manual/de/function.str-starts-with.php